test/controllers/IncapsulaControllerTest.ts

```diff
@@ -0,0 +1,362 @@
+import * as Lab from '@hapi/lab';
+import * as nock from 'nock';
+import * as request from 'supertest';
+import { server } from '../../src/darkstar';
+export const lab = Lab.script();
+const { describe, it, beforeEach } = lab;
+
+describe('/v1/caches/incapsula', () => {
+  let incapsulaAPIMock: nock.Scope;
+
+  beforeEach(() => {
+    incapsulaAPIMock = nock('https://my.incapsula.com');
+  });
+
+  describe('/zones/${zone_id}', () => {
+    describe('DELETE', () => {
+      let flushRequest: request.Test;
+      let incapsulaFlushMock: nock.Interceptor;
+
+      beforeEach(() => {
+        flushRequest = request(server.listener)
+          .delete('/v1/caches/incapsula/zones/abcd')
+          .set('accept', 'application/json');
+        incapsulaFlushMock = incapsulaAPIMock.post(
+          '/api/prov/v1/sites/cache/purge?api_id=1234&api_key=4321&site_id=abcd'
+        );
+      });
+
+      it('should flush a complete zone of the Incapsula cache', () => {
+        incapsulaFlushMock.matchHeader('accept', 'application/json').reply(200, { status: 'ok' });
```

Comment:
Incapsula replies something like : `{"res":0,"res_message":"OK","debug_info":{"id-info":"13008"}}` There is no key `status`.

Reply:
👍